client/src/components/ChartVisualization.jsx

```diff
@@ -0,0 +1,105 @@
+import React from 'react';
+import * as d3 from 'd3';
+
+import './../styles/ChartVisualization.scss';
+//  _________ .__                   __ ____   ____.__                    .__  .__                __  .__               
+//  \_   ___ \|  |__ _____ ________/  |\   \ /   /|__| ________ _______  |  | |__|____________ _/  |_|__| ____   ____  
+//  /    \  \/|  |  \\__  \\_  __ \   __\   Y   / |  |/  ___/  |  \__  \ |  | |  \___   /\__  \\   __\  |/  _ \ /    \ 
+//  \     \___|   Y  \/ __ \|  | \/|  |  \     /  |  |\___ \|  |  // __ \|  |_|  |/    /  / __ \|  | |  (  <_> )   |  \
+//   \______  /___|  (____  /__|   |__|   \___/   |__/____  >____/(____  /____/__/_____ \(____  /__| |__|\____/|___|  /
+//          \/     \/     \/                              \/           \/              \/     \/                    \/ 
+
+
+class ChartVisualization extends React.Component {
+
+  componentDidMount() {
+    const fullWidth = this.props.fullWidth;
+    const fullHeight = this.props.fullHeight;
+
+    const margins = { top: 15, right: 10, bottom: 30, left: 50 };
+    const width = fullWidth - margins.right - margins.left;
+    const height = fullHeight - margins.top - margins.bottom;
+
+    const allDates = this.props.dataset.map(datum => datum.date);
+
+    const svg = d3.select(this.svgElement)
+      .attr('width', fullWidth)
+      .attr('height', fullHeight)
+      .append('g')
+        .attr('transform', `translate(${margins.left}, ${margins.top})`);
```

Comment:
should this be aligned with previous ?

Reply:
With the way this is chained, the `.append('g')` on line 28 creates a new element. All subsequent lines are now dealing with this new element, not the previous `svg` element, so I indented line 29 to indicate this.